index.js

```diff
@@ -18,56 +19,81 @@
  * December 2013
  */
 
-module.exports = Lash
-
 function Lash (spec) {
 
-    var self = {}
-    if (!spec) spec = {}
+    var self = {};
+    if (!spec) spec = {};
+
+    self.startTime = null;
+
+    self.startTimer = function () {
+        self.startTime = process.hrtime();
+    };
+
+    self.time = function () {
+        var stackTime = process.hrtime(self.startTime);
+        var numSeconds = stackTime[0] + stackTime[1] / 1e9;
+        return numSeconds;
+    };
 
     Object.keys(spec).forEach( function (key) {
-        self[key] = spec[key]
-    })
+        self[key] = spec[key];
+    });
 
-    var steps = []
+    var steps = [];
 
 
     function stack (fn) {
-        steps.push(fn)
+        steps.push(fn);
     }
 
     function collapse () {
-        var args = Array.prototype.slice.call(arguments)
-          , cb_ = steps.pop()
-          , n = 0
-          , l = steps.length
-
-        nextStep(lashNext, args)
-
-        function lashNext () {
-            var args = Array.prototype.slice.call(arguments)
-            var er = args.shift()
-            if (er)
-                return cb_.apply(self, [er].concat(args))
-            if (++n === l)
-                return cb_.apply(self, [null].concat(args))
-
-            return nextStep(lashNext, args)
-        }
+        var args = Array.prototype.slice.call(arguments);
+        var cb_ = steps.pop();
+        var n = 0;
+        var l = steps.length;
 
         function nextStep (cb, args) {
-            var s = steps[n]
+            var s = steps[n];
+
+            if (typeof s === 'function') {
+                return s.apply(self, [cb].concat(args));
+            } else {
+                throw new Error('Only functions in chain!: ' + s);
+            }
+        }
 
-            if (typeof s === "function")
-                return s.apply(self, [cb].concat(args))
+        var lashNext = function (args) {
 
-            else throw new Error("Only functions in chain!: "+s)
+            var er = args.shift();
+
+            if (er) {
+                lashNext = function () {};
+                return cb_.apply(self, [er].concat(args));
+            }
+
+            if (++n === l) {
+                lashNext = function () {};
```

Comment:
some comments on here, this code was woefully uncommented to begin with, lets improve it some.